src/mono/netcore/System.Private.CoreLib/src/System/Environment.iOS.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using NSSearchPathDirectory = Interop.Sys.NSSearchPathDirectory;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static Dictionary<SpecialFolder, string> s_specialFolders;
+
+        private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOption option)
+        {
+            if (s_specialFolders == null)
+            {
+                Interlocked.CompareExchange(ref s_specialFolders, new Dictionary<SpecialFolder, string>(), null);
+            }
+
+            string path;
+            lock (s_specialFolders)
+            {
+                if (!s_specialFolders.TryGetValue(folder, out path))
+                {
+                    path = GetSpecialFolder(folder);
+                    s_specialFolders[folder] = path;
+                }
+            }
+            return path;
+        }
+
+        private static string GetSpecialFolder(SpecialFolder folder)
+        {
+            switch (folder)
+            {
+                // TODO: fix for tvOS (https://github.com/dotnet/runtime/issues/34007)
+                // The "normal" NSDocumentDirectory is a read-only directory on tvOS
+                // and that breaks a lot of assumptions in the runtime and the BCL
+
+                case SpecialFolder.Personal:
+                case SpecialFolder.LocalApplicationData:
+                    return Interop.Sys.SearchPath(NSSearchPathDirectory.NSDocumentDirectory);
```

Comment:
The native function may return NULL in some cases, in which case SearchPath would return null.  Is that possible here?

Reply:
Ouch, indeed, thanks.